github.go

```diff
@@ -91,6 +93,7 @@ func NewGitHubClient(source Source) (*GitHubClient, error) {
 			v4URL = source.GitHubAPIURL + "graphql"
 		}
 		clientV4 = githubv4.NewEnterpriseClient(v4URL, httpClient)
+		isEnterprise = true
```

Comment:
this is basically saying once `souce.GitHubAPIURL` is provided then it is an enterprise github deployment. Is that true? Could a github deployment be public but not enterprise version?

Reply:
in this one case I belive if you are providing something what is not default github api, we should consider this as github enterprise but I may be wrong there

Reply:
at least `isEnterprise = true` should be set whenever `NewEnterpriseClient` is called

Reply:
this is what is happening, when we have case with custom url, we are assuming  that enterprise client will be used